dist/s_style

```diff
@@ -13,24 +14,25 @@ fi
 
 # s_style is applied to every source file by default. It has a fast mode where it only runs against
 # modified files, according to git. Use -F for fast mode.
-search_str="find bench examples ext src test -name '*.[ch]' -o -name '*.in' -o -name '*.dox' \
-    -o -name '*.cxx' -o -name '*.cpp' -o -name '*.py' -o -name '*.cmake' -o -name '*.i'"
 single_file=0
-if [ $# -eq 1 ]; then
-    if [ "$1" = "-F" ]; then
-        check_fast_mode
-        search_str="git diff --name-only $(last_commit_from_dev) | grep -E '\.(c|h|cpp|cxx|i|py|dox|cmake)$'"
-    else
-        single_file=1
+if [[ "$#" -eq 1 && "$1" != "-F" ]]; then
+    single_file=1
+else
+    if [[ -n "$NO_RECURSE" ]]; then
```

Comment:
Why have we added the `NO_RECURSE` check? Also please update "INTERNAL ERROR!" below to describe what the error we're seeing is

Reply:
The issue here is that the script calls itself with a list of files. If for some reason the callee doesn't recognise this intent, it will find all files and call itself again. It can lead to an infinite recursion with exponential blow up in the number of processes.
This is an attempt to prevent it. I know there is a better way of doing it but that's probably a subject to a separate work.

I updated the comment to be more descriptive.